receptorctl: Remove setuptools runtime dependency

dsavineau opened the pull request:
Due to the `pkg_resources` import then setuptools is a runtime dependency.
The `pkg_resources` module is deprecated and we should use `importlib.metadata` instead which is a python builtin (since 3.8)

https://setuptools.pypa.io/en/latest/pkg_resources.html
https://docs.python.org/3.9/library/importlib.metadata.html

dsavineau committed:
receptorctl: Remove setuptools runtime dep

Due to the pkg_resources import then setuptools is a runtime
dependency.
The pkg_resources module is deprecated and we should use importlib.metadata
instead which is a python builtin (since 3.8)

https://setuptools.pypa.io/en/latest/pkg_resources.html
https://docs.python.org/3.9/library/importlib.metadata.html

Signed-off-by: Dimitri Savineau <dsavinea@redhat.com>

AaronH88 commented:
Thanks for the PR @dsavineau ,

If I understand correctly this module is not available in python 3.7?

If so, please feel free to drop 3.7 support from our CI pipe [here](https://github.com/ansible/receptor/blob/devel/.github/workflows/pull_request.yml#L110) as part of this PR and Ill be happy to merge

dsavineau committed:
workflows: Remove python 3.7 from matrix

Python 3.7 is EOL and since we're using importlib.metadata which
is only included starting 3.8 then dropping that release from
the test matrix.

Signed-off-by: Dimitri Savineau <dsavinea@redhat.com>

AaronH88 reviewed (approved):
lgtm
